READY: Completely reorganizes the /obj/ folder

SomeAngryMiner opened the pull request:
i pray to any and all gods and paranormal beings people dont make things conflict with this creature.

kills most "universal" icon files under the /obj/ icon tree, favoring folders and separate DMIs for each thing for ease of finding and editing by multiple people.

Adds error icons for when an icon state or whatever is not shown

eventually makes naming consistent across all things (see: Sleeper.dm being capitalized while everything else isn't, or bodyscannerconsole.dm being converted to body_scanner_console in a similar fashion)

separates machine states into overlays and emissives for consistency and, well, glowy bits.

erases unused icons wherever found (i might have missed some), except obvious legacy stuff that has been moved to _legacy .dmis

this is still ongoing so expect some things to be atomized even more

kills the Robotic_fabricator which was just a weird prosthetic printer that looked exactly like an exofab

big thanks to  @l3lb0t aka melnye aka Karl for helping me with this sysphean task

clothing nor mobs are touched currently because those are massive PRs on their own

![image](https://github.com/Baystation12/Baystation12/assets/53237389/006f87d4-ee44-4ad3-adab-cfaac612de64)


:cl: SomeAngryMiner, l3lb0t
rscadd: most machines now have emissive lights.
rscadd: most machines also have maintenance panels, if they were missing them.
rscadd: thermal regulators now have overlays when cooling.
/:cl:

SierraKomodo reviewed (changes requested):
You've merged dev into your branch, which will cause some shenanigans. Squash, rebase, resolve any conflicts it screams about, and only do it through rebasing when you need to update the branch in the future.

SomeAngryMiner commented:
icondiff aside, all conflicts have been solved, and what remains are compiler timeouts.

Will keep as a draft for a few more days as i add error icons to every file and continue converting machinery to the new standard.

SierraKomodo reviewed (changes requested):
Test failures:

> FAIL 344 spawn uses (expecting exactly 345)
> FAIL 397 attackby() override (expecting exactly 398)

SomeAngryMiner commented:
> Test failures:
> 
> > FAIL 344 spawn uses (expecting exactly 345)
> > FAIL 397 attackby() override (expecting exactly 398)

Those are because robotic_fabricator is gone completely, which housed both of those: it was deprecated, and unused. Effectively a prosthetic organ printer with the visuals of an exofab.

SomeAngryMiner commented:
Ready for review, but still needs a few tweaks before merge.

Added error icons to all overviewed files, and removed all duplicate files i could find, hence the increased file count.

once its ready for merge i heavily. HEAVILY. Recommend a test go first to see if i missed any icons. Such icons should be identifiable by the new error sign.

![image](https://github.com/Baystation12/Baystation12/assets/53237389/27f54411-80ec-40a9-8ae0-fc9d36c39c02)

SomeAngryMiner commented:
i should also note this should make any old icon errors resurface, so more cleaning up might be in place.

SierraKomodo reviewed (dismissed):
Some quick things I noticed with broken sprites or overlays/underlays in testing:
- Meat Fridge (`/obj/structure/closet/secure_closet/freezer/meat`)
- Chairs (`/obj/structure/bed/chair`)
- Bar Stool (`/obj/item/stool/bar/padded`)
- Holopad (`/obj/machinery/hologram/holopad`)
- Catwalks (`/obj/structure/catwalk `)
- Unpowered fire alarms (`/obj/machinery/firealarm`)
- Table frame (`/obj/structure/table`)

There's probably more but these are what immediately came out to me when I loaded example and torch maps and zoomed around.

juxjux9930 commented:
Adding to sierra's findings
- `/obj/structure/closet/secure_closet/freezer/fridge` is also an error icon, so it's probably an issue with `/obj/structure/closet/secure_closet/freezer` or something further up. Clicking on error'd fridges makes the correct icons show, however.
- `/obj/item/seeds/algaeseed`, `/obj/item/seeds/gukhe`, and potentially `/obj/item/seeds/angelmycelium ` all have the error icon as their little logo. I say potentially about the last one because I can't quite make out what is on it.
- `/obj/machinery/vending` and all the subtypes I tested are overlayed by the error icon when they dispense something.
- `/obj/machinery/fabricator` doesn't animate anymore when printing, and `/obj/machinery/r_n_d/protolathe`'s animation seems delayed when it starts and stops.
- `/obj/machinery/r_n_d/destructive_analyzer` disappears except for the overlays when anything is inserted into it, but reappears when the object is deconstructed or removed.
- `/obj/item/reagent_containers/dropper` is invisible, with no error icon.

l3lb0t commented:
> * `/obj/structure/closet/secure_closet/freezer/fridge` is also an error icon, so it's probably an issue with `/obj/structure/closet/secure_closet/freezer` or something further up. Clicking on error'd fridges makes the correct icons show, however.

Secure closets in general are just messed up. Biowaste carts also error.

SomeAngryMiner commented:
update log:
- Droppers fixed
- lathes fixed.
- fridges and lockers will have their error icons removed for the moment, since fixing that is extremely more invasive.
- holopads fixed
- all sources of light fixed.

continuing to work on this tonight.

SomeAngryMiner commented:
and, with this last push, i can say with some certainty it is ready to go for a testmerge.

Only thing that's missing is other people to find bugs, and for us to fix 'em.

SomeAngryMiner commented:
thanks to joint efforts with @l3lb0t , we've cleared every issue with the branch.
Waiting until tomorrow for the merge.

Only thing to note is that the grab icon error was effectively unidentifiable on local.
It literally never mentions icons.

We spent over 2 hours looking for that exact same error with 4 different people with no results.

Then at the final fixing compile it decided to stop appearing. For good. Without any further interaction given to it.

I think we should be scared, but, hey, its done.

SomeAngryMiner commented:
final files pushed. Ready for testmerge number dos.

SomeAngryMiner committed:
reorganizes obj folder

SomeAngryMiner commented:
the slowest internet ever disconnected my local with the remote when pushing, resulting in an empty commit.

SomeAngryMiner commented:
merge conflicts fixed.

SierraKomodo commented:
Test merge be live again.
